components/tidb_query_vec_expr/src/impl_compare.rs

```diff
@@ -303,28 +339,66 @@ pub fn greatest_time(ctx: &mut EvalContext, args: &[Option<BytesRef>]) -> Result
     Ok(greatest.map(|time| time.to_string().into_bytes()))
 }
 
+#[rpn_fn(nullable, varg, min_args = 2, capture = [ctx])]
+#[inline]
+pub fn least_time(mut ctx: &mut EvalContext, args: &[Option<BytesRef>]) -> Result<Option<Bytes>> {
+    let mut least = Some(Time::parse_from_i64(
+        &mut ctx,
+        MAX_TIMESTAMP,
+        TimeType::DateTime,
+        0,
+    )?);
+    for arg in args {
+        match arg {
+            Some(arg_val) => {
+                let s = match str::from_utf8(arg_val) {
+                    Ok(s) => s,
+                    Err(err) => {
+                        return ctx
+                            .handle_invalid_time_error(Error::Encoding(err))
+                            .map(|_| Ok(None))?;
+                    }
+                };
+                match Time::parse_datetime(ctx, &s, Time::parse_fsp(&s), true) {
+                    Ok(t) => least = min(least, Some(t)),
+                    Err(_) => {
+                        return ctx
+                            .handle_invalid_time_error(Error::invalid_time_format(&s))
+                            .map(|_| Ok(None))?;
+                    }
+                }
+            }
+            None => {
+                return Ok(None);
+            }
+        }
+    }
+
+    Ok(least.map(|time| time.to_string().into_bytes()))
+}
+
 #[inline]
-fn do_get_extremum<T, E>(args: &[Option<&T>], chooser: E) -> Result<Option<T>>
+fn do_get_extremum<'a, T, E>(args: &[Option<&'a T>], chooser: E) -> Result<Option<T::Owned>>
 where
-    T: Ord + Copy,
-    E: Fn(T, T) -> T,
+    T: Ord + ToOwned + ?Sized,
+    E: Fn(&'a T, &'a T) -> &'a T,
 {
     let first = args[0];
     match first {
         None => Ok(None),
         Some(first_val) => {
-            let mut res = *first_val;
+            let mut res = first_val;
             for arg in &args[1..] {
                 match arg {
                     None => {
                         return Ok(None);
                     }
                     Some(v) => {
-                        res = chooser(res, **v);
+                        res = chooser(res, *v);
                     }
                 }
             }
-            Ok(Some(res))
+            Ok(Some(res.to_owned()))
         }
     }
 }
```

Comment:
Some tests?

Reply:
rest LGTM

Reply:
Maybe macros in https://github.com/tikv/tikv/pull/8418/files/c3013ccebf09b5ea1c3127aa455610b55804d749#diff-2632cab92125451435bb96885bd52017R321 will be used in copr-tests? But im not so sure

Reply:
Unit tests I mean

Reply:
Yes, we should add some private function unit tests to this help function, at least specializing the `T` to an integertype.